sdk/loadtestservice/Azure.Developer.LoadTesting/src/Models/Test.Serialization.cs

```diff
@@ -0,0 +1,124 @@
+    {
+        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            if (Optional.IsDefined(PassFailCriteria))
+            {
+                writer.WritePropertyName("passFailCriteria"u8);
+                writer.WriteObjectValue(PassFailCriteria);
+            }
+            if (Optional.IsCollectionDefined(Secrets))
+            {
+                writer.WritePropertyName("secrets"u8);
+                writer.WriteStartObject();
+                foreach (var item in Secrets)
+                {
+                    writer.WritePropertyName(item.Key);
+                    writer.WriteObjectValue(item.Value);
+                }
+                writer.WriteEndObject();
+            }
+            if (Optional.IsDefined(Certificate))
+            {
+                writer.WritePropertyName("certificate"u8);
+                writer.WriteObjectValue(Certificate);
+            }
+            if (Optional.IsCollectionDefined(EnvironmentVariables))
+            {
+                writer.WritePropertyName("environmentVariables"u8);
+                writer.WriteStartObject();
+                foreach (var item in EnvironmentVariables)
+                {
+                    writer.WritePropertyName(item.Key);
+                    writer.WriteStringValue(item.Value);
+                }
+                writer.WriteEndObject();
+            }
+            if (Optional.IsDefined(LoadTestConfiguration))
+            {
+                writer.WritePropertyName("loadTestConfiguration"u8);
+                writer.WriteObjectValue(LoadTestConfiguration);
+            }
+            if (Optional.IsDefined(Description))
+            {
+                writer.WritePropertyName("description"u8);
+                writer.WriteStringValue(Description);
+            }
+            if (Optional.IsDefined(DisplayName))
+            {
+                writer.WritePropertyName("displayName"u8);
+                writer.WriteStringValue(DisplayName);
+            }
+            if (Optional.IsDefined(SubnetId))
+            {
+                writer.WritePropertyName("subnetId"u8);
+                writer.WriteStringValue(SubnetId);
+            }
+            if (Optional.IsDefined(KeyvaultReferenceIdentityType))
+            {
+                writer.WritePropertyName("keyvaultReferenceIdentityType"u8);
+                writer.WriteStringValue(KeyvaultReferenceIdentityType);
+            }
+            if (Optional.IsDefined(KeyvaultReferenceIdentityId))
+            {
+                writer.WritePropertyName("keyvaultReferenceIdentityId"u8);
+                writer.WriteStringValue(KeyvaultReferenceIdentityId);
+            }
+            writer.WriteEndObject();
+        }
+
+        internal void WritePatch(Stream stream)
+        {
+            _element.WriteTo(stream, 'P');
+        }
+
+        internal static Test DeserializeTest(JsonElement element)
```

Comment:
My understanding is patch operation only impacts serialization process. What is the purpose we change the deserialization process? For example, `op createOrUpdate(@body body: Test): Test;` I think the returned payload should fully respect how `Test` looks like.

Reply:
I believe I did it this way because I had changed the internal deserialization constructor to take a MutableJsonElement, and this made it easy to implement using this approach.  I can't remember if it was required, or I was interested in exploring what it looked like this way.  One thought I had is that, since the standard generated deserialization routine creates a lot of local variables, we might be able to reduce allocations with this approach.  We would need to measure performance to understand whether it improved performance or not, though.

Reply:
What I want to express is, for example, `Test` has a required property like `id`. The `id` is changed to optional because the `Test` is in a patch. Then when a `Test` is returned from service, `id` should be back to required, right? In that case we should deserialize `Test` in normal way not in this patch way. I mean we should throw error when there is no `id` in the payload.

Reply:
If the implementation of the model type changes, then we aren't able to deserialize it in the normal way - we need to rely on the new implementation.

For example, look at the updated [Test.Deserialize method here](https://github.com/Azure/azure-sdk-for-net/pull/37341/files#diff-3c001ef92069e9641fc3ee83fae6abb55cb3243fa3d7418c8a681b38fb45db61R118-R123).  Since `id` is retrieved from the private `_element` field in Test now, `_element` must be populated in the deserialization routine.  The new implementation of Deserialize does this, by calling through to Test's internal deserialization constructor that takes a MutableJsonElement.

Reply:
But new implementation doesn't throw error if `id` is missing from payload, which means we change our existing behavior.

Reply:
I don't think we are changing existing behavior, because, as I understand it, we are not currently generating convenience methods for PATCH APIs - only protocol methods.  If you know of existing behavior that we might be breaking, please share it so we have that as reference.